library/x509_crt.c

```diff
@@ -2681,6 +2680,13 @@ static int x509_inet_pton_ipv4(const char *src, void *dst)
             if (digit > 9) {
                 break;
             }
+
+            /* Don't allow leading zeroes. These might mean octal format,
+             * which this implementation does not support. */
+            if (octet == 0 && num_digits > 0) {
+                break;
```

Comment:
I might be nitpicking but why not just return `-1` here? That would seem clearer.

Reply:
I wanted to stay with the pre-existing _style_: one exit point from the function with a clear condition for success: exactly four octets parsed, NUL at the end. 
Before I looked at it closer I would say that it's a matter of preference, but now that I noticed that in a test case of `10.10.10.01` the error is indicated by the fact that `p` doesnt point at `\0` but the last `1` instead (the code breaks before `p++`) - the behaviour is not that obvious. 
I'll change the function to clearly indicate an exit on each of the errors.